moz/l10n/properties/serialize.py

```diff
@@ -0,0 +1,100 @@
+# Copyright Mozilla Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections.abc import Callable, Iterator
+from typing import Literal
+
+from translate.storage.properties import propunit
+
+from ..resource import Entry, M, Metadata, Resource, V
+
+
+def properties_serialize(
+    resource: Resource[V, M],
+    encoding: Literal["iso-8859-1", "utf-8", "utf-16"] = "utf-8",
+    serialize_message: Callable[[V], str] | None = None,
+    serialize_metadata: Callable[[Metadata[M]], str | None] | None = None,
+    trim_comments: bool = False,
+) -> Iterator[str]:
+    """
+    Serialize a resource as the contents of a .properties file.
+
+    Section identifiers will be prepended to their constituent message identifiers.
+    Multi-part message identifiers will be joined with `.` between each part.
```

Comment:
What is an example of a multi-part message identifier?

Reply:
Identifiers are lists of strings, so something like `['foo', 'bar']`. Parsing .properties will always result in identifiers with length 1, but if you then post-process the resource to e.g. split an identifier like `['foo.bar']` into `['foo', 'bar']` you'll get some. They also show up in resources parsed from Fluent, where we have ids like `['name', 'attr']`.